oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/ElasticIndexDefinition.java

```diff
@@ -381,4 +408,238 @@ protected IndexDefinition createInstance(NodeState indexDefnStateToUse) {
             return new ElasticIndexDefinition(root, indexDefnStateToUse, indexPath, indexPrefix);
         }
     }
+
+    /**
+     * Represents the inference configuration for an Elasticsearch index definition.
+     * This class holds the properties and queries used for inference.
+     */
+    public static class InferenceDefinition {
+
+        /**
+         * List of properties used for inference.
+         */
+        public List<Property> properties;
+
+        /**
+         * List of queries used for inference.
+         */
+        public List<Query> queries;
+
+        public InferenceDefinition() { }
+
+        /**
+         * Constructs an InferenceDefinition from a given NodeState.
+         *
+         * @param inferenceNode the NodeState containing the inference configuration
+         */
+        public InferenceDefinition(NodeState inferenceNode) {
+            if (inferenceNode.hasChildNode("properties")) {
+                this.properties = StreamSupport.stream(inferenceNode.getChildNode("properties").getChildNodeEntries().spliterator(), false)
+                        .map(cne -> new Property(cne.getName(), cne.getNodeState()))
+                        .collect(Collectors.toList());
+            }
+            if (inferenceNode.hasChildNode("queries")) {
+                this.queries = StreamSupport.stream(inferenceNode.getChildNode("queries").getChildNodeEntries().spliterator(), false)
+                        .map(cne -> new Query(cne.getName(), cne.getNodeState()))
+                        .collect(Collectors.toList());
+            }
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            InferenceDefinition that = (InferenceDefinition) o;
+            return Objects.equals(properties, that.properties) && Objects.equals(queries, that.queries);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(properties, queries);
+        }
+
+        /**
+         * Represents a property used for inference.
+         */
+        public static class Property {
+            /**
+             * The name of the property. It will be prefixed with {@link ElasticIndexDefinition#INFERENCE}. when stored in the index.
```

Comment:
Is the `.` part of the prefix? Initially I thought the `.` before `when` was a typo in the sentence. Maybe put `{@link ElasticIndexDefinition#INFERENCE}.` in double quotes.